.github/workflows/lint.yml

```diff
@@ -43,3 +43,21 @@ jobs:
 
       - name: Run tests
         run: go test ./...
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Nix
+        uses: DeterminateSystems/nix-installer-action@main
+
+      - name: Use Nix development environment
+        uses: rrbutani/use-nix-shell-action@v1
+        with:
+          devShell: .#ci
```

Comment:
This isn't needed for `nix build`; I only used it for the golangci-lint environment.

It takes some time to eval the shell and fetch the packages, and idk if we have a limit on CI time.